src/extensions/default/JavaScriptCodeHints/main.js

```diff
@@ -859,6 +859,62 @@ define(function (require, exports, module) {
         }
 
         /*
+         * Handle Rename menu/keyboard command.
+         */
+        function handleRename() {
+            var offset,
+                handleFindRefs;
+
+
+            // Only provide rename when cursor is in JavaScript content
+            if (!session || session.editor.getModeForSelection() !== "javascript") {
+                return null;
+            }
+
+            var result = new $.Deferred();
+
+            /**
+             * Make a find ref request.
+             * @param {Session} session - the session
+             * @param {number} offset - the offset of where to jump from
+             */
+            function requestFindRefs(session, offset) {
+                var response = ScopeManager.requestFindRefs(session, session.editor.document, offset);
+
+                if (response.hasOwnProperty("promise")) {
+                    response.promise.done(handleFindRefs).fail(function () {
```

Comment:
Can we use `then` and `catch` to align the old Deferred syntax with the more modern Promises/A+ syntax?